email-common/src/main/java/io/commercetools/sunrise/email/EmailSender.java

```diff
@@ -1,6 +1,82 @@
+ * <p>Instances of the e-mail service are created with a configuration such that they can later send messages
+ * without requiring further credentials or configuration of servers.
+ * <p>The service API is based on the Java Mail API. Note that implementations of this service might choose not to
+ * <em>send</em> e-mails using the (default implementation of the) Java Mail API but to merely accept messages created
+ * using the Java Mail API. You may want to consult the <a href="https://java.net/projects/javamail/pages/Home">samples,
+ * JavaDocs and specification of the Java Mail API</a>.
+ */
+@FunctionalInterface
 public interface EmailSender {
 
-    // TODO
+    /**
+     * Create a completion stage that asynchronously sends an e-mail using the configuration of this e-mail
+     * service. Before this method returns, the message author argument of this method is passed an empty
+     * {@link MimeMessage} that the {@link EmailSender} created. The message author shall prepare the message
+     * so it can be send. Sending will happen asynchronously, though.
+     * <p>
+     * Refer to the <a href="https://javamail.java.net/nonav/docs/api/">Java Mail JavaDoc</a> and the
+     * <a href="http://javamail.java.net/nonav/docs/JavaMail-1.5.pdf">Java Mail Specification</a> on how to configure
+     * instances of {@link MimeMessage}. The following code shows a simple example.
+     * <pre>{@code
+     * CompletionStage<String> completionStage = emailSender.send(msg -> {
+     *      msg.setFrom("foo@domain.com");
+     *      msg.setSubject("Subject, "UTF-8");
+     *      msg.setText("Text", "UTF-8");
+     *      msg.setSentDate(new Date());
+     *      msg.setRecipients(Message.RecipientType.TO, "bar@domain.com");
+     * });
+     * }</pre>
+     * The {@link CompletionStage} returned by this method may be combined with further completion stages. To directly
+     * send the email and get its message ID, obtain a completable future as in the following example.
+     * <pre>{@code
+     * String messageID = completionStage.toCompletableFuture().get();
+     * }</pre>
+     * Checked exceptions that occur while sending the email are instances of {@link EmailSenderException} that are
+     * wrapped in an {@link ExecutionException} thrown by {@link CompletableFuture#get()}.
+     * <p>To send e-mails synchronously, use {@link #sendAndWait(MessageAuthor, int, TimeUnit)}.
+     *
+     * @param messageAuthor a message author that fills the empty message created by the email sender.
+     *                      Note that {@link MimeMessage} instances are not immutable. Messages passed to the message
+     *                      author must only be used by that single author instance and must not be passed elsewhere.
+     * @return A completion stage for sending the message asynchronously.
+     * @see #sendAndWait(MessageAuthor, int, TimeUnit)
+     */
+    @Nonnull
+    CompletionStage<String> send(@Nonnull final MessageAuthor messageAuthor);
```

Comment:
it also be directly

```
@Nonnull
CompletionStage<String> send(@Nonnull final Consumer<MimeMessage> messageAuthor);
```

Since MessageAuthor is a functional interface without default methods it is easier to understand the
send signature with inlining the type.

With reading MessageAuthor in the signature without knowing the type caused confusion since I want to send a mail and an author is just a part of it.